nixos/modules/services/matrix/dendrite.nix

```diff
@@ -271,11 +276,6 @@ in
         LimitNOFILE = 65535;
         EnvironmentFile = lib.mkIf (cfg.environmentFile != null) cfg.environmentFile;
         LoadCredential = cfg.loadCredential;
-        ExecStartPre = ''
-          ${pkgs.envsubst}/bin/envsubst \
-            -i ${configurationYaml} \
-            -o /run/dendrite/dendrite.yaml
-        '';
```

Comment:
If you make this a list instead of a string, than its possible to expand the list by defining your own lists in your own config:

```
ExecStartPre = [''
  ${pkgs.envsubst}/bin/envsubst \
    -i ${configurationYaml} \
    -o /run/dendrite/dendrite.yaml
''];
```

I would prefer this approach because it saves one additional nix build (with expensive sandbox setup) and it makes the unit file more readable by not having additional scripts to look at.